### PR TITLE
Participant database

### PR DESCRIPTION
**Pull request by jparisu**
Merge after #41

### CODE REVIEW

File: include/ddsrouter/participant/ParticipantDatabase.hpp [hunk 1]
```diff
@@ -50,6 +52,8 @@ class ParticipantDatabase
 protected:
 
     std::map<ParticipantId, std::shared_ptr<IParticipant>> participants_;
+
+    mutable std::shared_timed_mutex mutex_;
```
Comment (juanlofer-eprosima) [inline include/ddsrouter/participant/ParticipantDatabase.hpp:56]: Would perhaps a shared_mutex be enough?

File: src/cpp/participant/ParticipantDatabase.cpp [hunk 2]
```diff
@@ -33,27 +32,49 @@ void ParticipantDatabase::add_participant(
...
 
 std::shared_ptr<IParticipant> ParticipantDatabase::get_participant(
         const ParticipantId& id) const
 {
-    // TODO
-    throw UnsupportedException("ParticipantDatabase::get_participant not supported yet");
+    std::shared_lock<std::shared_timed_mutex> lock(mutex_);
+    auto it = participants_.find(id);
+
+    if (it != participants_.end())
```
Comment (juanlofer-eprosima) [inline src/cpp/participant/ParticipantDatabase.cpp:53]:
```suggestion
    if (it == participants_.end())
```

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]

File: include/ddsrouter/participant/ParticipantDatabase.hpp (hunk 1 above)
Comment (jparisu) [reply include/ddsrouter/participant/ParticipantDatabase.hpp:56]: `shared_mutex` is not added until C++17
https://en.cppreference.com/w/cpp/thread/shared_mutex

File: src/cpp/participant/ParticipantDatabase.cpp (hunk 2 above)
Comment (jparisu) [reply src/cpp/participant/ParticipantDatabase.cpp:53]: :+1:

**Review by juanlofer-eprosima — APPROVED** [review]
LGTM

File: include/ddsrouter/participant/ParticipantDatabase.hpp [hunk 3]
```diff
@@ -20,6 +20,8 @@
 #define _DDSROUTER_PARTICIPANT_PARTICIPANTDATABASE_HPP_
 
 #include <map>
+#include <mutex>
```
Comment (juanlofer-eprosima) [inline include/ddsrouter/participant/ParticipantDatabase.hpp:23]:
```suggestion
```

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]
NIT

**Review by juanlofer-eprosima — APPROVED** [review]
LGTM

**Pull request merged**